base/src/org/adempiere/process/YearEndClosing.java

```diff
@@ -0,0 +1,186 @@
+		closingAccountId = CreateAccount(getElementValueId());
+		if (closingAccountId == 0)
+			return "";
+		MJournalBatch journalBatch = new MJournalBatch(getCtx(), 0, get_TrxName());	
+		
+			
+		BigDecimal balance = Env.ZERO;
+		String documentno = Msg.translate(Env.getCtx(), "Closing")+ "_"  + getDateAcct().toString();
+		String description =  Msg.translate(Env.getCtx(), "Closing") + " "  + getDateAcct().toString();
+		journalBatch.setAD_Org_ID(getOrgId());
+		journalBatch.setDateAcct(getDateAcct());
+		journalBatch.setDateDoc(getDateAcct());
+		journalBatch.setDescription(documentno);
+		journalBatch.setC_DocType_ID(getDocTypeId());
+		journalBatch.setDocumentNo(description);
+		journalBatch.setGL_Category_ID(getCategoryId());
+		journalBatch.setControlAmt(Env.ZERO);
+		journalBatch.setC_Currency_ID(acctSchema.getC_Currency_ID());
+		journalBatch.save();
+		
+		String whereclause = "accounttype in ('E', 'R')  and ISSUMMARY = 'N' AND c_Element_ID=?";
+		int elementId = DB.getSQLValueEx(get_TrxName(), 
+				"SELECT C_Element_ID FROM C_AcctSchema_Element WHERE elementtype = 'AC' AND C_AcctSchema_ID=?", getAcctSchemaId());
+		List<MElementValue> accounts = new Query(getCtx(), MElementValue.Table_Name, 
+				whereclause, get_TrxName())
+		.setClient_ID()
+		.setParameters(elementId)
+		.setOrderBy("value")
+		.list();		
+		for (MElementValue el : accounts) 
+		{			
+			balance = SearchBalance(el.getC_ElementValue_ID());
+			if (balance.equals(Env.ZERO))
+				continue;
+			CreateJournal(el.getC_ElementValue_ID(), journalBatch, acctSchema, balance);
+		}
+		return "@Created@: " + journalBatch.getDocumentNo();		
+	}	//	doIt
+	
+	private int CreateAccount(int elementValueId)
+	{   
+		
+		
+		MAccount acc = MAccount.get(getCtx(), getAD_Client_ID(),
+		getOrgId(), getAcctSchemaId(), 
+		elementValueId, 0,
+		0, 0, 0, 
+		0, 0, 0, 
+		0, 0, 0,
+		0, 0, 0,0,0,0, get_TrxName());
+		if (acc == null)
+			return 0;
+		return acc.get_ID();
+		
+	}
+	
+	private Boolean CreateJournal(int elementValueID, MJournalBatch journalBatch, 
+			MAcctSchema acctSchema, BigDecimal balance)
+	{
+		MJournal journal = new MJournal(journalBatch);
+		int accountId = CreateAccount(elementValueID);
+		MElementValue elementValue = new MElementValue(getCtx(), elementValueID, get_TrxName() );
+		
+		journal.setC_Currency_ID(acctSchema.getC_Currency_ID());
+		journal.setC_AcctSchema_ID(acctSchema.get_ID());
+		journal.setC_ConversionType_ID(MConversionType.getDefault(getAD_Client_ID()));
+		journal.setDescription(Msg.translate(Env.getCtx(), "Closing") + " " + elementValue.getValue() + " " + elementValue.getName());
+		journal.setDocumentNo(DB.getDocumentNo(getAD_Client_ID(), MJournal.Table_Name, get_TrxName()));
+		journal.setGL_Category_ID(getCategoryId());
+		journal.save();
+		if (!CreateLines(accountId, journal, elementValueID, balance))
+			return false;				
+		return true;
+	}
+	
+	private Boolean CreateLines(int accountID, MJournal journal, 

```

Comment:
@SusanneCalderon Hi Susanne, @marcalwestf Mario, Can you change the name of the methods to Camel format?, the methods inside a class start with a lowercase?

```private Boolean createLines(int accountId, MJournal journal, ```